tfx/components/transform/executor.py

```diff
@@ -146,8 +141,7 @@ def __init__(self, file_pattern: Text,
       materialize_output_path: The file path where to write the dataset.
     """
     self._file_pattern = file_pattern
-    file_pattern_suffix = os.path.join(
-        *file_pattern.split(os.sep)[-self._FILE_PATTERN_SUFFIX_LENGTH:])
+    file_pattern_suffix = hashlib.sha256(file_pattern.encode()).hexdigest()
```

Comment:
Would it be possible to prefix it with the file_pattern_suffix that was used before?
Also, let's rename this variable to dataset_identifier now.

Reply:
Updated. Now uses file_patter_suffix as prefix, with sha256 hash suffix.